timemachine/potentials/nonbonded.py

```diff
@@ -562,6 +562,93 @@ def coulomb_interaction_group_energy(q_ligand: Array, q_prefactors: Array) -> fl
 
 
 # utilities for efficiently recomputing energy as a function of ligand LJ parameters
+
+# note: two cases:
+# (1) only varying LJ eps parameters (can summarize using 1 float per ligand atom)
+# (2) simultaneously varying LJ eps and sig parameters (summarized using 20 floats per ligand atom)
+
+
+# case (1) : only varying ligand LJ eps parameters
+def lj_eps_prefactor_on_atom(x_i, x_others, sig_i, sig_others, eps_others, box=None, cutoff=1.2) -> float:
+    """Precompute part of (sum_j LennardJones(x_i, x_j; (sig_i, eps_i), (sig_j, eps_j))) that does not depend on eps_i
+
+    Parameters
+    ----------
+    x_i : [D] array
+        position of focus atom (in ligand)
+    x_others: [N_env, D] array
+        positions of all other atoms (in environment)
+    sig_i: float
+        LJ_sig of focus atom
+    sig_others: [N_env] array
+        LJ_sig of all other atoms (in environment)
+    eps_others: [N_env] array
+        LJ_eps of all other atoms (in environment)
+    box: optional diagonal [D, D] array
+    cutoff: float
+
+    Returns
+    -------
+    prefactor_i : float
+        sum_j 4 * sqrt(eps_j) * ((sig_ij/r_ij)**12 - (sig_ij/r_ij)**6)
```

Comment:
sqrt(eps_j) or just eps_j?

Reply:
adjusted in https://github.com/proteneer/timemachine/pull/1212/commits/40ee31c9a647c1a1b4cc687dcf6cd2a5024f0be9